lib/widgets/home.dart

```diff
@@ -1,97 +1,539 @@
+  const _MentionsButton();
+
+  @override
+  IconData get icon => ZulipIcons.at_sign;
+
+  @override
+  String label(ZulipLocalizations zulipLocalizations) {
+    return zulipLocalizations.mentionsPageTitle;
+  }
+
+  @override
+  void onPressed(BuildContext context) {
+    Navigator.of(context).push(MessageListPage.buildRoute(
+      context: context, narrow: const MentionsNarrow()));
+  }
+}
+
+class _StarredMessagesButton extends _MenuButton {
+  const _StarredMessagesButton();
+
+  @override
+  IconData get icon => ZulipIcons.star;
+
+  @override
+  String label(ZulipLocalizations zulipLocalizations) {
+    return zulipLocalizations.starredMessagesPageTitle;
+  }
+
+  @override
+  void onPressed(BuildContext context) {
+    Navigator.of(context).push(MessageListPage.buildRoute(
+      context: context, narrow: const StarredMessagesNarrow()));
+  }
+}
+
+class _CombinedFeedButton extends _MenuButton {
+  const _CombinedFeedButton();
+
+  @override
+  IconData get icon => ZulipIcons.message_feed;
+
+  @override
+  String label(ZulipLocalizations zulipLocalizations) {
+    return zulipLocalizations.combinedFeedPageTitle;
+  }
+
+  @override
+  void onPressed(BuildContext context) {
+    Navigator.of(context).push(MessageListPage.buildRoute(
+      context: context, narrow: const CombinedFeedNarrow()));
+  }
+}
+
+class _ChannelsButton extends _NavigationBarMenuButton {
+  const _ChannelsButton({required super.tabNotifier});
+
+  @override
+  IconData get icon => ZulipIcons.hash_italic;
+
+  @override
+  String label(ZulipLocalizations zulipLocalizations) {
+    return zulipLocalizations.channelsPageTitle;
+  }
+
+  @override
+  HomePageTab get navigationTarget => HomePageTab.channels;
+}
+
+class _DirectMessagesButton extends _NavigationBarMenuButton {
+  const _DirectMessagesButton({required super.tabNotifier});
+
+  @override
+  IconData get icon => ZulipIcons.user;
+
+  @override
+  String label(ZulipLocalizations zulipLocalizations) {
+    return zulipLocalizations.recentDmConversationsPageTitle;
+  }
+
+  @override
+  HomePageTab get navigationTarget => HomePageTab.directMessages;
+}
+
+class _MyProfileButton extends _MenuButton {
+  const _MyProfileButton();
+
+  @override
+  IconData? get icon => null;
+
+  @override
+  Widget buildLeading(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    return Avatar(userId: store.selfUserId, size: 24, borderRadius: 4);
+  }
+
+  @override
+  String label(ZulipLocalizations zulipLocalizations) {
+    return zulipLocalizations.profilePageTitle;
+  }
+
+  @override
+  void onPressed(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    Navigator.of(context).push(
+      ProfilePage.buildRoute(context: context, userId: store.selfUserId));
+  }
+}
+
+class _SwitchAccountButton extends _MenuButton {
+  const _SwitchAccountButton();
+
+  @override
+  // TODO(design): choose an icon
+  IconData get icon => ZulipIcons.globe;
```

Comment:
Yeah, not `globe` because that's the icon we use to mean "web-public channel".

The TODO can remain, but pending that it'd be better to just make this blank — have `buildLeading` return `SizedBox.shrink()`.